01_fibonacci/main.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+
+	for num := 0; num < 10; num++ {
+		fmt.Fprintln(out, printNumber(num))
+	}
+
+}
+func printNumber(x int) int {
```

Comment:
Another recursive one! Bonus points if you can state the time complexity.

Reply:
well its bad than iterative..

iterative will be O(n) and this will be O(2^n). I just did it anyways.

I don't get the bonus point i know  :(